content/en/contribute/code/repository-checklist.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "Repository Checklist"
+linkTitle: "Repository Checklist"
```

Comment:
Maybe "Repository Creation Checklist" is better?

Reply:
This is a great point, thanks @garethbowen! We should consider a rewrite.

However, I think I have a more nascent concern for both product and repositories: We should always strive to help a real customer/solve a real problem as early as possible.  For example, if [CHT Monitoring](https://github.com/medic/cht-monitoring), which I would consider a product, needed to meet all the "at least do the following" items, we wouldn't be live now and it may be weeks or months until we met all the requirements on that list.  I don't think the investment in checking all those boxes would be worth if it because A) the lost time on helping actual customers and B) we may have a failed product in 3 months and the month we spent doing the checklist was a waste.

The list is actually great though!  I think we can do the rewrite Gareth suggested and greatly reduce the "must haves" and introduce a  "nice to have" list maybe?  The must have list might be:

* The main branch is locked via [branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule).
* Merges are done through PRs.
* PRs reference related issues.
* Commit formats follow the [guidelines](http://localhost:1313/contribute/code/workflow/#commits).
* Secrets are not part of the commit history or made public.
* For public repositories the main branch should contain a `license` and `readme.md`
*  `main` branch is always shippable.
*  A minimum number of reviewers (usually 2) for a PR merge is enforced by policy